azure-pipelines.yml

```diff
@@ -31,6 +46,20 @@ stages:
         ptf_name: ptf_vms6-1
         tbtype: t0
         vmtype: ceos
+        section: part-2
+
+  - job:
+    displayName: "kvmtest-t0"
+    timeoutInMinutes: 400
+    dependsOn:
+      - t0-part1
+      - t0-part2
+    condition: |
```

Comment:
I guess the condition is always().
You can define a variable to indicate the job is successful.
```
variables:
  DEP_RESULT: ${{ convertToJson(dependencies.*.result) }}
steps:
  script: |
    ### exist successful or failed based on $(DEP_RESULT)
```

Reply:
Thanks for the reminder. always is a good way, we can also add continueOnError: true in two sub jobs to let job kvmtest-t0 run every time even sub job fails.
Conditionally run a step such as the following code doesn't work: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#conditional-insertion
```
     steps:
     - ${{ if and(eq(variables.resultOfPart1, 'Succeeded'), eq(variables.resultOfPart2, 'Succeeded')) }}:
       - script: 'echo "Run kvmtest-t0 successfully!"'
     - ${{ else }}:
	- script: 'echo "Run kvmtest-t0 failed!"; exit 1'
```

So change to use script:
```
    steps:
    - script: |
        if [ $(resultOfPart1) == "Succeeded" ] && [ $(resultOfPart2) == "Succeeded" ]; then
          echo "Both job kvmtest-t0-part1 and kvmtest-t0-part2 are passed."
          exit 0
        else
          echo "Either job kvmtest-t0-part1 or job kvmtest-t0-part2 failed! Please check the detailed information."
          exit 1
        fi
```